> @rbuisson-invn thoughts?
To be honest I do not even understand the issue here, nor the fix :)
I do not see any conversion from enum to boolean in the expression :

/__w/zephyr/zephyr/drivers/sensor/tdk/icp201xx/icp201xx_drv.c:219:70: error: converting the enum constant to a boolean [-Werror,-Wint-in-bool-context]
        if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS ||
                                                                                                                                         ^

== operator should have precedence over || operator.
Sorry but, I would be glad if you could clarify this for me.

--- a/drivers/sensor/tdk/icp201xx/icp201xx_drv.c
+++ b/drivers/sensor/tdk/icp201xx/icp201xx_drv.c
@@ -216,8 +216,8 @@ static int icp201xx_channel_get(const struct device *dev, enum sensor_channel ch
 {
 	struct icp201xx_data *data = (struct icp201xx_data *)dev->data;
 
-	if (chan != SENSOR_CHAN_AMBIENT_TEMP && chan != SENSOR_CHAN_PRESS &&
-	    chan != SENSOR_CHAN_ALTITUDE) {
+	if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS ||
+	      chan == SENSOR_CHAN_ALTITUDE)) {
 		return -ENOTSUP;
 	}
 	icp201xx_mutex_lock(dev);

